scripts/nlp_language_modeling/build_knn_map_index.py

```diff
@@ -110,19 +160,55 @@ def get_tokenizer(args):
     return tokenizer
 
 
-def process_sentence_chunks(ds: MMapRetrievalIndexedDataset, tokenizer, chunk_size: int):
+def calculate_start_end(total_chunks, total_shards, shard_id):
+    shard_size = total_chunks // total_shards
+    splits = list(range(0, total_chunks, shard_size))
+    if shard_id < total_shards - 1:
+        start = splits[shard_id]
+        total_chunks = splits[shard_id + 1]
+    elif shard_id == total_shards - 1:
+        start = splits[shard_id]
+        total_chunks = total_chunks
+    else:
+        raise ValueError(f'{shard_id} bigger than {total_shards}')
+    return start, total_chunks
+
+
+def process_sentence_chunks(
```

Comment:
Can you add a docsring describing what the function?

Reply:
added